Library/Homebrew/test/os/mac/keg_spec.rb

```diff
@@ -5,28 +5,65 @@
 
   subject { described_class.new(keg_path) }
 
-  describe "#mach_o_files" do
-    let(:keg_path) { HOMEBREW_CELLAR/"a/1.0" }
+  let(:keg_path) { HOMEBREW_CELLAR/"a/1.0" }
+  let(:file) { keg_path/"lib/i386.dylib" }
 
-    before(:each) { (keg_path/"lib").mkpath }
-
-    after(:each) { subject.unlink }
+  before(:each) do
+    (keg_path/"lib").mkpath
+    cp dylib_path("i386"), file
+  end
 
-    it "skips hardlinks" do
-      cp dylib_path("i386"), keg_path/"lib/i386.dylib"
-      ln keg_path/"lib/i386.dylib", keg_path/"lib/i386_hardlink.dylib"
+  after(:each) { subject.unlink }
 
+  describe "#change_dylib_id" do
+    before(:each) do
       subject.link
-      expect(subject.mach_o_files.count).to eq(1)
     end
 
-    it "isn't confused by symlinks" do
-      cp dylib_path("i386"), keg_path/"lib/i386.dylib"
-      ln keg_path/"lib/i386.dylib", keg_path/"lib/i386_hardlink.dylib"
-      ln_s keg_path/"lib/i386.dylib", keg_path/"lib/i386_symlink.dylib"
+    it "does nothing if given id is same as file's dylib id" do
+      id = file.dylib_id
+      subject.change_dylib_id(id, file)
+      expect(file.dylib_id).to eq(id)
+    end
+
+    it "changes file's dylib id to given id" do
+      id = "blah.B.dylib"
+      file.ensure_writable do
+        subject.change_dylib_id(id, file)
+      end
+      expect(file.dylib_id).to eq(id)
+    end
+  end
 
+  describe "#change_install_name" do
+    before(:each) do
       subject.link
-      expect(subject.mach_o_files.count).to eq(1)
+    end
+
+    it "does nothing if given name is same as file's install name" do
+      initial_name = ""
```

Comment:
You shouldn't need to pre-declare `initial_name` here.

Reply:
I was initially using `initial_name` in a different block from the one where it was declared. Hence, I was pre-declaring it. I'm using the expect assertion inside the same block now.